### PR TITLE
Issue 85: Add get_all_hosts/vcenters & register_vm functions

### LINKED ISSUE #221: Incorrect string join on `get_vsphere_vm` & `get_sql_db` trailing `&` in query
**Expected Behavior**

Incorrect string join on `get_vsphere_vm` & `get_sql_db` trailing `&` in query

**Current Behavior**

When querying for either a vsphere vm or a sql db a trailing `&` is added to each query. This is not required and can be removed from the query

**Failure Information (for bugs)**

image

**Steps to Reproduce**

```py
rubrik_cdm.Connect(enable_logging=True)
rubrik.get_vsphere_vm(name="jbrasser-win")
```

### PR DESCRIPTION
**Pull request by jaapbrasser**
# Description

Added additional functionality in the form of `get_all_hosts` / `get_all_vcenters` functions

## Related Issue

Resolve #85 
Resolve #221 

## Motivation and Context

Adds additional functionality

## How Has This Been Tested?

* Please describe in detail how you tested your changes.
* Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **[CONTRIBUTION](https://github.com/rubrikinc/rubrik-sdk-for-python/blob/master/CONTRIBUTING.md)** document.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by jaapbrasser** [comment]
Revived stale branch, fixed bugs and added unit tests.

File: rubrik_cdm/data_management.py [hunk 1]
```diff
@@ -2258,3 +2254,39 @@ def get_esxi_subnets(self, timeout=15):  # pylint: ignore
...
+    def get_all_hosts(self, timeout=15):
+            """Retrieve information for each host connected to the Rubrik cluster.
+            Arguments:
+                timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {15})
+            Returns:
+                dict -- The result of the API call `GET /v1/host`
+            """
+
+            self.log('get_all_hosts: Getting information for each host on the Rubrik cluster.')
+            current_hosts = self.get('v1', '/host', timeout=timeout)
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py:2267]: we can just return the API call directly

`return = self.get('v1', '/host', timeout=timeout)`

File: rubrik_cdm/data_management.py [hunk 2]
```diff
@@ -1913,9 +1913,7 @@ def get_vsphere_vm(self, name=None, is_relic=None, effective_sla_domain_id=None,
             if ((key == 'limit') or (key == 'offset')) and not isinstance(value, int):
                 raise InvalidParameterException('The limit and offset paremeter must be an integer')
 
-        query = ''
-        for key, value in parameters.items():
-            query = query + ("{}={}".format(key, value) + '&')
+        query = '&'.join(['%s=%s' % kv for kv in parameters.items()])
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py:1916]: Can we add a comment describing what this line is doing?

File: rubrik_cdm/data_management.py [hunk 3]
```diff
@@ -2062,9 +2060,7 @@ def get_sql_db(self, db_name=None, instance=None, hostname=None, availability_gr
...
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py:2063]: Can we add a comment describing what this line is doing?

File: rubrik_cdm/data_management.py (hunk 1 above)
Comment (jaapbrasser) [reply rubrik_cdm/data_management.py]: Updated code, thanks for spotting that!

New code in: cf7e991

File: rubrik_cdm/data_management.py (hunk 2 above)
Comment (jaapbrasser) [reply rubrik_cdm/data_management.py:1917]: Added comments for the string join

New code in: cf7e991

File: rubrik_cdm/data_management.py (hunk 3 above)
Comment (jaapbrasser) [reply rubrik_cdm/data_management.py:2065]: Added comments for the string join

New code in: cf7e991

**Pull request merged**